rust/db_handling/src/identities.rs

```diff
@@ -96,6 +97,168 @@ lazy_static! {
     static ref REG_PATH: Regex = Regex::new(r"^(?P<path>(//?[^/]+)*)(///(?P<password>.+))?$").expect("known value");
 }
 
+#[derive(Clone, Encode, Decode, Debug, Eq, PartialEq)]
+pub enum ExportAddrs {
+    V1(ExportAddrsV1),
+}
+
+#[derive(Clone, Encode, Decode, Debug, Eq, PartialEq)]
+pub struct ExportAddrsV1 {
+    pub addrs: Vec<SeedInfo>,
+}
+
+#[derive(Clone, Encode, Decode, Debug, Eq, PartialEq)]
+pub struct SeedInfo {
+    /// Name of the seed.
+    pub name: String,
+
+    /// Public key of the root key.
+    ///
+    /// `None` if user has created no root keys.
+    pub multisigner: Option<MultiSigner>,
+
+    /// Derived keys.
+    pub derived_keys: Vec<AddrInfo>,
+}
+
```

Comment:
@pgolovkin the structures have been updated to show hierarchical information, that should suffice your usecases